### ISSUE TITLE
Add Flower Baseline: FedPFT

### ISSUE THREAD

**Issue by mahdibeit** [adap/flower#3232]
### Paper

Mahdi Beitollahi, Alex Bie, Sobhan Hemati, Leo Maxime Brunswic, Xu Li, Xi Chen, Guojun Zhang (2024). Parametric Feature Transfer: One-shot Federated Learning with Foundation Models

### Link

https://arxiv.org/abs/2402.01862

### Maybe give motivations about why the paper should be implemented as a baseline.

This paper introduces FedPFT (Federated Learning with Parametric Feature Transfer), a method that utilizes the transferability of foundation models to enhance both accuracy and communication efficiency in one-shot FL. FedPFT brings foundation models to one-shot FL without requiring users to train or transmit large foundation models to the server.

![image](https://github.com/adap/flower/assets/96784135/c5406941-668d-4062-bb65-30284970a51a)

### Is there something else you want to add?

This baseline surpasses previous one-shot SOTA methods. I think it can be very useful for the Flower community to utilize large, pre-trained models. 

I will be implementing the main experiment of the paper (Sec. for 5.2) using 50 clients, the Caltech101 dataset, and Clip-ViT as the feature extractor. 

### For first time contributors

- [x]  Read the [`first contribution` doc](https://flower.ai/docs/first-time-contributors.html)
- [x]  Complete the Flower tutorial
- [x]  Read the Flower Baselines docs to get an overview:
    - [x]  [How to use Flower Baselines](https://flower.ai/docs/baselines/how-to-use-baselines.html)
    - [x]  [How to contribute a Flower Baseline](https://flower.ai/docs/baselines/how-to-contribute-baselines.html)

### Prepare - understand the scope

- [X] Read the paper linked above
- [X] Decide which experiments you'd like to reproduce. The more the better!
- [x] Follow the steps outlined in [Add a new Flower Baseline](https://flower.ai/docs/baselines/how-to-contribute-baselines.html#add-a-new-flower-baseline).
- [x] You can use as reference [other baselines](https://github.com/adap/flower/tree/main/baselines) that the community merged following those steps.

### Verify your implementation

- [x] Follow the steps indicated in the `EXTENDED_README.md` that was created in your baseline directory
- [x] Ensure your code reproduces the results for the experiments you chose
- [x] Ensure your `README.md` is ready to be run by someone that is no familiar with your code. Are all step-by-step instructions clear?
- [x] Ensure running the formatting and typing tests for your baseline runs without errors.
- [x] Clone your repo on a new directory, follow the guide on your own `README.md` and verify everything runs.

**Comment by jafermarq** [comment]
Hi @mahdibeit , this paper looks cool! Let me know if you need of any help to start with the baseline creation process. If some of the steps aren't clear, ping me!

**Comment by mahdibeit** [comment]
Hi @jafermarq, I just wanted to let you know that I have opened a pull request #3268. Let me know if it requires any changes.

**Comment by jafermarq** [comment]
Hi @mahdibeit, thank you for the ping. I've approved the workflow that runs the CI tests (one of which is for the tests that check your baseline). I'll review your code soon

**Comment by jafermarq** [comment]
@mahdibeit , many thanks for adding your `FedPFT` into the [Flower Baselines](https://flower.ai/docs/baselines/)!! Would you be interested in presenting it in an upcoming Flower Monthly event? It runs on the first Wednesday of each month. Would the June session (June 5th) work for you? More info in our [Flower Monthly page](https://flower.ai/conf/flower-monthly/). Feel free to reach to me via the Flower Slack! (https://flower.ai/join-slack/)
